### PR TITLE
M282 - Detach Servo

### PR DESCRIPTION
**Pull request by Dakkaron**
### Description

As detailled in issue #22757, I have trouble with RC servos and an LPC1768 based board. The servo signal jitters, which causes the servo to hum and run hot. This PR adds a GCode (M282) which allows the user to detach servos whenever they want. It fixes the problem that I have, but it might be useful on other boards and in other scenarios as well.

With this, I can disable the servo after a tool change and thus stop it from humming and running hot. I already lost one servo to overheating.

DEACTIVATE_SERVOS_AFTER_MOVE cannot be used in my use case, since my servo is too weak to hold it's position during the tool change when detached.

So my current scenario is, I use M280 to use the servo in the right position for the tool change (M280 automatically attaches a servo), then I move the print head to perform the tool change, then I use M282 to detach the servo. This keeps it nice, quiet and cool. 

### Requirements

This PR requires a Servo

### Benefits

Allows for manually detaching servos, keeping them powered as long as necessary, and keeping them detached, quiet and cool, whenever they are not needed.

### Configurations

Nothing special, you just need a servo and enable the servo with `#define NUM_SERVOS 1`

### Related Issues

#22757

### CODE REVIEW

**Comment by thinkyhead** [comment]
Before this can be added to the firmware it must be added to the RepRap.org wiki page so it will not end up used by other firmware for something else. I see there that the `M281` G-code is not used, so why did you choose `M282`, out of curiosity?

**Comment by thinkyhead** [comment]
Although this is a modest addition, it may still be best to make it optional for the benefit of tiny AVR boards with very little flash space. That has been the customary approach when adding new code that will necessarily increase the base build size.

File: Marlin/src/gcode/control/M282.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+void GcodeSuite::M282() {
+
+  if (!parser.seen('P')) return;
+
+  const int servo_index = parser.value_int();
+  if (WITHIN(servo_index, 0, NUM_SERVOS - 1)) {
+    DETACH_SERVO(servo_index);
+  }
+  else
+    SERIAL_ERROR_MSG("Servo ", servo_index, " out of range");
```
Comment (thinkyhead) [inline Marlin/src/gcode/control/M282.cpp]: Note that by default OctoPrint disconnects from the printer when it sees "Error:". If this isn't a fatal error then it may be fine to use `ECHO` instead.

**Comment by Dakkaron** [comment]
> Before this can be added to the firmware it must be added to the RepRap.org wiki page so it will not end up used by other firmware for something else. I see there that the `M281` G-code is not used, so why did you choose `M282`, out of curiosity?

M281 is already used by Marlin to edit servo angles (https://marlinfw.org/docs/gcode/M281.html)

> Although this is a modest addition, it may still be best to make it optional for the benefit of tiny AVR boards with very little flash space. That has been the customary approach when adding new code that will necessarily increase the base build size.

Will do

File: Marlin/src/gcode/control/M282.cpp (hunk 1 above)
Comment (Dakkaron) [reply Marlin/src/gcode/control/M282.cpp]: Makes sense, will do. Thanks for the review!

**Comment by Dakkaron** [comment]
Everything requested is done, including the RepRap.org Wiki page. Please let me know if there is anything else that needs to be done!

**Comment by p3p** [comment]
This is not a comment on this feature, disabling the servo signal on demand has its uses.

As for the bug you are describing if you are using a hardware PWM pin on an LPC176x the timing has practically no jitter (at least that's my observation with a signal analyser), however if the pin you use is attached to a hardware PWM channel that is already in use (it is shared between a few [pins](https://github.com/p3p/pio-framework-arduino-lpc176x)), or is not a hardware PWM pin, it will use the software fallback mode that uses interrupts, that fallback signal can be jittery by nature.

**Comment by thinkyhead** [comment]
> `M281` is already used by Marlin…

So it is! I've just added it to the RepRap wiki so it won't be missed.

**Comment by thinkyhead** [comment]
Made some adjustments and updated the Configurations and MarlinDocumentation repos to prepare for deployment.

File: platformio.ini [hunk 2]
```diff
@@ -87,7 +87,7 @@ default_src_filter = +<src/*> -<src/config> -<src/HAL> +<src/HAL/shared>
   -<src/lcd/touch/touch_buttons.cpp>
   -<src/sd/usb_flashdrive/lib-uhs2> -<src/sd/usb_flashdrive/lib-uhs3>
   -<src/sd/usb_flashdrive/Sd2Card_FlashDrive.cpp>
-  -<src/sd/cardreader.cpp> -<src/sd/Sd2Card.cpp> -<src/sd/SdBaseFile.cpp> -<src/sd/SdFatUtil.cpp> -<src/sd/SdFile.cpp> -<src/sd/SdVolume.cpp> -<src/gcode/sd>
```
Comment (Dakkaron) [inline platformio.ini:90]: Was this change on purpose?

Comment (thinkyhead) [reply platformio.ini:90]: Yes, `-<src/gcode/sd>` was listed twice.

**Comment by thinkyhead** [comment]
Oops. We missed the fact that `M280 Pn S-1` already does servo detach, so the new code is not needed.

**Comment by Dakkaron** [comment]
Interesting. This behaviour is not documented in either the Marlinfw.org nor the reprap.org GCode documentations.

**Comment by thinkyhead** [comment]
It was applied by @tpruvot in #14085 (May 2019) but was never properly documented.
